backend/services/employerService.js

```diff
@@ -1,11 +1,50 @@
 const employerDAO = require('../DAO/employerDAO');
 
-module.exports = {
+const SuperError = require('../errors').SuperError;
+const ERR_BAD_REQUEST = require('../errors').ERR_BAD_REQUEST;
+
+const locationService = require('../services/locationService');
+const domainOfActivityService = require('../services/domainOfActivityService');
+
+
+
+module.exports = self = {
     getEmployer: (id) => {
         return employerDAO.getEmployer(id);
     },
 
     getLastEmployer: async () => {
         return await employerDAO.getLastEmployer();
+    },
+
+    createNewEmployer: async (details) => {
+        if(details && typeof(details.location.country) == 'string' && typeof(details.location.city) == 'string' && typeof(details.name) == 'string' && typeof(details.description) == 'string' && !isNaN(details.domainOfActivityId)){
+            let check = await domainOfActivityService.existsDomainOfActivity(details.domainOfActivityId);
+            if(check){
+                let newEmployer = await employerDAO.addNewEmployer(details);
+                if(details.location){
+                    newEmployer.location = await self.saveEmployerLocation(newEmployer.id, details.location);
```

Comment:
I have noticed some problems if we use the same naming ("self") to represent the service inside itself. Can you check that we don't use "self" in any other component? if we do, can we change this to something else?

Reply:
I changed it to employerService